packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -16,6 +17,10 @@ const PAGINATION_ERROR_MESSAGES = {
 	NO_DATA_FOUND: 'No Submitted data found',
 } as const;
 
+// A constant representing a list separator using the pipe character (|),
+// which cannot be used in entity names due to its illegality.
+const entitySeparator = '|' as const;
```

Comment:
For a future PR (this one is fine as is), I'd make this a fallback value within the function, while still allowing to pass custom separators